src/nifreeze/utils/iterators.py

```diff
@@ -33,14 +33,17 @@ def linear_iterator(size: int | None = None, **kwargs) -> Iterator[int]:
 
     Parameters
     ----------
-    size : :obj:`int`
-        Number of volumes in the dataset
-        (for instance, the number of orientations in a DWI)
+    size : :obj:`int` or ``None`` (optional)
```

Comment:
size : :obj:`int` or ``None`` (optional)
```suggestion
    size : :obj:`int` or ``None``, optional
```